pkg/registry/controller/etcd/etcd_test.go

```diff
@@ -57,125 +55,102 @@ func createController(storage *REST, rc api.ReplicationController, t *testing.T)
 	return *newRc, nil
 }
 
-var validPodTemplate = api.PodTemplate{
-	Template: api.PodTemplateSpec{
+func validController() *api.ReplicationController {
+	return &api.ReplicationController{
 		ObjectMeta: api.ObjectMeta{
-			Labels: map[string]string{"a": "b"},
+			Name:      "foo",
+			Namespace: api.NamespaceDefault,
 		},
-		Spec: api.PodSpec{
-			Containers: []api.Container{
-				{
-					Name:            "test",
-					Image:           "test_image",
-					ImagePullPolicy: api.PullIfNotPresent,
+		Spec: api.ReplicationControllerSpec{
+			Selector: map[string]string{"a": "b"},
+			Template: &api.PodTemplateSpec{
+				ObjectMeta: api.ObjectMeta{
+					Labels: map[string]string{"a": "b"},
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:            "test",
+							Image:           "test_image",
+							ImagePullPolicy: api.PullIfNotPresent,
+						},
+					},
+					RestartPolicy: api.RestartPolicyAlways,
+					DNSPolicy:     api.DNSClusterFirst,
 				},
 			},
-			RestartPolicy: api.RestartPolicyAlways,
-			DNSPolicy:     api.DNSClusterFirst,
 		},
-	},
-}
-
-var validControllerSpec = api.ReplicationControllerSpec{
-	Selector: validPodTemplate.Template.Labels,
-	Template: &validPodTemplate.Template,
-}
-
-var validController = api.ReplicationController{
-	ObjectMeta: api.ObjectMeta{Name: "foo", Namespace: "default"},
-	Spec:       validControllerSpec,
+	}
 }
 
 func TestCreate(t *testing.T) {
 	storage, fakeClient := newStorage(t)
 	test := resttest.New(t, storage, fakeClient.SetError)
+	controller := validController()
+	controller.ObjectMeta = api.ObjectMeta{}
```

Comment:
CreateTests had been written in a way (this wasn't me actually), that they are modifying parts of ObjectMeta (and they are assuming that it's empty). We can fix this, but I don't think we should do it in this PR.
To explain why this changed, before what we were passing to TestCreate was an object without ObjectMeta, not when I changed it to validController() we need to clear the ObjectMeta, because it contains them.